### DISCUSSION TITLE
Pipelining

### DISCUSSION THREAD

**Discussion by hermanbanken** [uptrace/bun#711] [Q&A]
In HTTP and RESP (Redis) pipelining makes a tremendous difference in throughput achievable with a connection.

In SQL, this can be done with a batching, such as [`pgx.Batch`](https://pkg.go.dev/github.com/jackc/pgx#Batch). However, in a ORM like Bun it might be difficult to model this.

Can pipelining be used with Bun, or is this out of scope?

[![https://upload.wikimedia.org/wikipedia/commons/thumb/1/19/HTTP_pipelining2.svg/1200px-HTTP_pipelining2.svg.png](https://upload.wikimedia.org/wikipedia/commons/thumb/1/19/HTTP_pipelining2.svg/1200px-HTTP_pipelining2.svg.png)](https://www.google.com/url?sa=i&url=https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FHTTP_pipelining&psig=AOvVaw2Tosub6g1kfif_yC-WJF9y&ust=1668775343842000&source=images&cd=vfe&ved=0CBEQjhxqFwoTCKib_N6etfsCFQAAAAAdAAAAABAD)

**Comment by imraan-go** [comment]
I think you are confusing batch query with connection pipelining. These are two different concept. Bun does not support query batching yet. But it does support bulk insert/update.

**Reply by hermanbanken** [reply]
Besides the name, what is the technical difference in the underlying connection in your understanding? AFAIK both do the exact same thing: emit multiple statements consecutively without waiting for individual answers.

The difference between those terms is how the client handles the responses, right?
